storage/memory/memory.go

```diff
@@ -70,6 +70,26 @@ func (r *Backend) Open(options url.Values) {
 	go r.maintenance()
 }
 
+func (r Backend) GetLastDataItems(tenant string, id string, numOfItems int) ([]storage.DataItem, error) {
+	res := make([]storage.DataItem,numOfItems)
+	r.checkID(tenant, id)
+	t, _ := r.tenant[tenant]
+	ts, _ := t.ts[id]
+	if len(ts.data) < numOfItems {
+		return nil, fmt.Errorf("Not Enough items in database: Found %d, Queried: %d", len(ts.data), numOfItems)
+	} else {
+		var idx int
+		for i := 0 ; i < numOfItems; i++ {
```

Comment:
since you're mapping `ts.data` into a different structure, I would suggest writing this loop differently:

```Go
for ix, dataT := range ts.data[len(ts.data)-numOfItems:] {
  res = append(res, storage.DataItem{
     Timestamp: dataT.timeStamp
     Value: dataT.Value})
}
```

Reply:
Nice! 👍